pkg/storage/tsdb/bucketindex/loader.go

```diff
@@ -144,6 +144,10 @@ func (l *Loader) GetIndex(ctx context.Context, userID string) (*Index, Status, e
 }
 
 func (l *Loader) cacheIndex(userID string, idx *Index, ss Status, err error) {
+	if errors.Is(err, context.Canceled) {
+		level.Info(l.logger).Log("msg", "skipping cache bucket index", "err", err)
```

Comment:
Umm, why do we need this log? And why we need to log it only if it is ctx canceled.
It feels weird to log the error when you already know it is ctx canceled.
Do we need `Error` level log here? Or maybe debug?

Reply:
I can remove the log. But i think there is no harm as well..

I put the error as we can in the future have other types of erros we dont wanna cache (ex: `context.DeadlineExceeded`) 

WTYT?